### PR TITLE
API: create an endpoint for fetching organizations #104

### PR DESCRIPTION
**Pull request by daxiousm**
# Description

Please provide a short summary explaining what this PR is about.

Fixes # 104(issue)

# How to test?

run "npm run dev"
then use swagger to test ,"http://localhost:3000/api/documentation/"

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/controllers/organizations.controller.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const HttpError = require('../lib/utils/http-error');
+const knex = require('../../config/db');
+
+const getOrganizations = async () => {
+  try {
+    return await knex('organizations').select('id', ' name', 'imageUrl');
```
Comment (tpetrina) [inline src/server/api/controllers/organizations.controller.js:6]: You have an error here in the last column.

File: src/server/api/controllers/organizations.controller.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+const getOrganizationsById = async (id) => {
+  try {
+    const organization = await knex('organizations')
+      .select('organizations.id as id', 'name', 'imageUrl')
```
Comment (tpetrina) [inline src/server/api/controllers/organizations.controller.js:15]: You have an error here in the last column.

File: src/server/api/controllers/organizations.controller.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      .where({ id });
+    if (organization.length === 0) {
+      throw new HttpError(
+        'Bad request',
+        `Cannot find organization with ID ${id}!`,
+        404,
+      );
+    }
+    return organization;
```
Comment (tpetrina) [inline src/server/api/controllers/organizations.controller.js:24]: You are returning a list here instead of just one item.

**Review by tpetrina — CHANGES_REQUESTED** [review]
The PR is not working, please fix. Good start btw.

**Review by Kiran-Serolla — COMMENTED** [review]
Hi Daxious,
I checked the api on swagger but it cannot fetch data. As Toni suggested, I think the last column should be setect(tablename.column name ) instead of just select(columnname)

File: src/server/api/controllers/organizations.controller.js [hunk 4]
```diff
@@ -1,33 +1,19 @@
...
   try {
-    return await knex('organizations').select('id', ' name', 'imageUrl');
-  } catch (err) {
-    console.log(err);
-  }
-};
+    const organizations = await knex('organizations').select(
+      'id',
+      'name',
+      'image_url',
```
Comment (tpetrina) [inline src/server/api/controllers/organizations.controller.js:8]: You need to return this back as `imageUrl`.

**Review by tpetrina — CHANGES_REQUESTED** [review]

File: src/server/api/controllers/organizations.controller.js (hunk 4 above)
Comment (daxiousm) [reply src/server/api/controllers/organizations.controller.js]: corrected as `organizations.image_url as imageUrl `

**Review by Kiran-Serolla — DISMISSED** [review]

**Review by tpetrina — DISMISSED** [review]

**Review by tpetrina — APPROVED** [review]
Good stuff

**Review by 1620prabn — APPROVED** [review]

**Pull request merged**